core/src/main/java/com/linecorp/armeria/internal/ArmeriaHttpUtil.java

```diff
@@ -868,7 +868,7 @@ public static void toNettyHttp1(
                 final AsciiString name = entry.getKey();
                 final String value = entry.getValue();
                 final AsciiString translatedName = translations.get(name);
-                if (translatedName != null) {
+                if (translatedName != null && !inputHeaders.contains(translatedName)) {
```

Comment:
is there any chance a header that validly has multiple values could end up here? maybe a comment saying that no input header is valid with multiple values.

Reply:
Added a comment on the translated headers that they should only contain headers that are present once in a request. Currently, the only translation is `:authority` -> `host` and maybe could just be hard-coded